src/main/java/app/linguistai/bmvp/service/gamification/StoreService.java

```diff
@@ -0,0 +1,206 @@
+package app.linguistai.bmvp.service.gamification;
+
+import app.linguistai.bmvp.exception.store.InsufficientUserItemQuantityException;
+import app.linguistai.bmvp.exception.NotFoundException;
+import app.linguistai.bmvp.exception.SomethingWentWrongException;
+import app.linguistai.bmvp.exception.currency.InsufficientGemsException;
+import app.linguistai.bmvp.exception.store.ItemNotEnabledException;
+import app.linguistai.bmvp.model.User;
+import app.linguistai.bmvp.model.enums.TransactionType;
+import app.linguistai.bmvp.model.gamification.store.StoreItem;
+import app.linguistai.bmvp.model.gamification.store.UserItem;
+import app.linguistai.bmvp.repository.IAccountRepository;
+import app.linguistai.bmvp.repository.gamification.store.IStoreItemRepository;
+import app.linguistai.bmvp.repository.gamification.store.IUserItemRepository;
+import app.linguistai.bmvp.response.gamification.store.RStoreItem;
+import app.linguistai.bmvp.response.gamification.store.RStoreItems;
+import app.linguistai.bmvp.response.gamification.store.RUserItem;
+import app.linguistai.bmvp.response.gamification.store.RUserItems;
+import app.linguistai.bmvp.service.currency.TransactionService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class StoreService {
+    private final IStoreItemRepository storeItemRepository;
+    private final IUserItemRepository userItemRepository;
+    private final IAccountRepository accountRepository;
+
+    private final TransactionService transactionService;
```

Comment:
Since the interface of this service already exists, it would be better to use `ITransactionService` here.

Reply:
changed